src/main/webapp/app/exam/participate/summary/result-overview/exam-result-overview.component.ts

```diff
@@ -89,11 +89,58 @@ export class ExamResultOverviewComponent implements OnInit, OnChanges {
         this.maxPoints = this.studentExamWithGrade?.maxPoints ?? 0;
         this.isBonusGradingKeyDisplayed = this.studentExamWithGrade.studentResult.gradeWithBonus?.bonusGrade != undefined;
 
-        this.overallAchievedPoints = this.studentExamWithGrade?.studentResult.overallPointsAchieved ?? 0;
-        this.overallAchievedPercentageRoundedByCourseSettings = roundScorePercentSpecifiedByCourseSettings(
-            (this.studentExamWithGrade.studentResult.overallScoreAchieved ?? 0) / 100,
-            this.studentExamWithGrade.studentExam?.exam?.course,
-        );
+        this.overallAchievedPoints = this.getOverallAchievedPoints();
+        this.overallAchievedPercentageRoundedByCourseSettings = this.getOverallAchievedPercentageRoundedByCourseSettings();
+    }
+
+    /**
+     * used as fallback if not pre-calculated by the server
+     */
+    private sumExerciseScores() {
+        let exerciseScoreSum = 0;
+
+        this.studentExamWithGrade.studentExam?.exercises?.forEach((exercise) => {
```

Comment:
maybe .reduce would be a better fit here

Reply:
Addressed with https://github.com/ls1intum/Artemis/pull/7588/commits/422f4aa5665a45f1f6e0254ffb16cedca492b201